pkg/profile/artifact.go

```diff
@@ -7,54 +7,180 @@ import (
 	"time"
 
 	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
 	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
 )
 
+type Status struct {
+	ResourcesExist     bool
+	NotReadyConditions []metav1.Condition
+}
+
 // CreateArtifacts creates and inserts objects to the cluster to deploy the
 // profile as a HelmRelease.
 func (p *Profile) CreateArtifacts(ctx context.Context) error {
-	if err := p.createGitRepository(ctx); err != nil {
+	gitRes, helmRes, kustomizeRes, err := p.makeArtifacts()
+	if err != nil {
+		return err
+	}
+
+	p.log.Info("creating GitRepository", "resource", gitRes.ObjectMeta.Name)
+	if err := p.client.Create(ctx, gitRes); err != nil {
 		return fmt.Errorf("failed to create GitRepository resource: %w", err)
 	}
 
-	switch kind := p.definition.Spec.Artifacts[0].Kind; kind {
```

Comment:
Interesting. I just read that we will support multiple ways for artifacts. Things like, raw yaml for Kubernetes resources and such. Or is that not this part of the code? :) I might be mistaken. I'm just wondering what prompted this refactor.

Reply:
this bit is just moving the existing code, it still exists in this file. Currently we only support 1 artifact, that can be of Kind Kustomize (which includes raw yaml) or Helm.

Reply:
Gotcha, thanks.